### PR TITLE
[13.0][MIG] stock_request_picking_type

### PR DESCRIPTION
**Pull request by ps-tubtim**
Standard Migration.

### CODE REVIEW

**Review by bodedra — APPROVED** [review]
Functional review. :+1:

**Comment by bodedra** [comment]
@marcelsavegnago @RLeeOSI Are you guys available to review this PR?

**Comment by bodedra** [comment]
ping @brian10048 

This will unlock FSM stock related v13 migration.

**Review by brian10048 — APPROVED** [review]

**Comment by bodedra** [comment]
ping @dreispt

**Review by nikul-serpentcs — APPROVED** [review]
Code review LGTM :+1:

**Comment by bodedra** [comment]
@OCA/logistics-maintainers ready to merge.

File: stock_request_picking_type/views/stock_request_order_views.xml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<odoo>
+    <record id="view_stock_request_order_filter" model="ir.ui.view">
```
Comment (rousseldenis) [inline stock_request_picking_type/views/stock_request_order_views.xml]: I'm wondering why this view is defined here and not in base module ???

File: stock_request_picking_type/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+{
+    "name": "Stock Request Picking Type",
+    "summary": "Add Stock Requests to the Inventory App",
+    "version": "13.0.1.0.0",
+    "license": "LGPL-3",
+    "website": "https://github.com/stock-logistics-warehouse",
+    "author": "Open Source Integrators, " "Odoo Community Association (OCA)",
+    "category": "Warehouse",
+    "depends": ["stock_request_submit"],
```
Comment (rousseldenis) [inline stock_request_picking_type/__manifest__.py]: Why depending on this module (apart for the search view - which IMHO should be defined in the base module) ?

IMHO, the subimtted state should be included in base module. @max3903 Any advice on this ?

**Review by rousseldenis — COMMENTED** [review]
Code review. Several questions about global architecture

File: stock_request_picking_type/__manifest__.py (hunk 2 above)
Comment (RLeeOSI) [reply stock_request_picking_type/__manifest__.py]: Agreed, stock_request_submit should be merged into stock_request. Its only function is to add the 'submitted' state to stock.request.order. This view should be merged with the searchview added in #903 .

Comment (bodedra) [reply stock_request_picking_type/__manifest__.py]: @ps-tubtim or @RLeeOSI would you improve this one.

Comment (ps-tubtim) [reply stock_request_picking_type/__manifest__.py]: I changed this code from `"depends": ["stock_request_submit"],` to `"depends": ["stock_request"],`.

File: stock_request_picking_type/views/stock_request_order_views.xml (hunk 1 above)
Comment (ps-tubtim) [reply stock_request_picking_type/views/stock_request_order_views.xml]: I deleted this search view and improve the search view of the core.

**Review by bodedra — APPROVED** [review]
LG :+1:

**Comment by bodedra** [comment]
@rousseldenis would you please update your review?

**Review by RLeeOSI — APPROVED** [review]
Looks good!

File: stock_request_picking_type/models/stock_request_order.py [hunk 3]
```diff
@@ -9,30 +9,25 @@ class StockRequestOrder(models.Model):
 
     @api.model
     def _get_default_picking_type(self):
+        companies = self._context.get("allowed_company_ids", []).copy()
```
Comment (rousseldenis) [inline stock_request_picking_type/models/stock_request_order.py:12]: @ps-tubtim Why having used this instead a search on companies ?

Comment (rousseldenis) [reply stock_request_picking_type/models/stock_request_order.py:12]: Moreover, I'm wondering why the security rule on picking type is not sufficent ?

https://github.com/odoo/odoo/blob/13.0/addons/stock/security/stock_security.xml#L75

**Pull request merged**